web/connectionHub.go

```diff
@@ -83,60 +106,120 @@ func handleRegistration(connection *models.UserConnection) {
 }
 
 const (
-	TEXT       = "TEXT"
-	SUBMIT     = "SUBMIT"
-	LOAD       = "LOAD"
-	GETRESULTS = "GETRESULTS"
+	Text       = "TEXT"
+	Submit     = "SUBMIT"
+	Load       = "LOAD"
+	GetResults = "GETRESULTS"
+	BanUser    = "BANUSER"
+	GetBanList = "GETBANLIST"
 )
 
 // We need to handle the state of the rooms and the q and a's
 func handleBroadcast(message models.UserResponse) {
-	message.Logger.Println("broadcast received:", message)
+	message.Logger.Println("broadcast received:", message.String())
 	room := rooms[message.RoomId]
+
 	switch strings.ToUpper(message.UserMessage.Action) {
-	case LOAD:
+	case Load:
 		room.Joinable = false
 		// since room is a copy, we have to assign it back to the map
 		rooms[message.RoomId] = room
+
 		if message.UserId == room.AdminId {
 			err := handleAdminLoad(message.RoomId, message.UserMessage.DeckId, message.Conn)
 			if err != nil {
-				Configs.Logger.Printf("there was an error: %#v", err)
+				Configs.Logger.Printf("there was an error: %#v\n", err)
 				_ = message.Conn.WriteJSON(models.NewErrorResponse(err.Error()))
 			}
 		} else {
-			Configs.Logger.Printf("%d is trying to access admin ", message.UserId)
+			Configs.Logger.Printf("%d is trying to access admin\n", message.UserId)
 			_ = message.Conn.WriteJSON(models.NewErrorResponse("non admin user"))
 		}
+	case BanUser:
+		handleAdminBanUser(&room, &message)
+		rooms[message.RoomId] = room
 
-	case TEXT:
+	case GetBanList:
+		if room.AdminId == message.UserId {
+			if banErr := message.Conn.WriteJSON(room.BannedList); banErr != nil {
+				Configs.Logger.Printf("there was an error: %#v", banErr)
+				_ = message.Conn.WriteJSON(models.NewErrorResponse(banErr.Error()))
+			}
+		} else {
+			Configs.Logger.Printf("%d is trying to access admin\n", message.UserId)
+			_ = message.Conn.WriteJSON(models.NewErrorResponse("non admin user"))
+		}
+
+	case Text:
 		handleTextMessage(message)
-	case SUBMIT:
+	case Submit:
 		if err := handleAnswerSubmissions(message); err != nil {
 			Configs.Logger.Printf("there was an error: %#v", err)
 			_ = message.Conn.WriteJSON(models.NewErrorResponse(err.Error()))
 		}
-	case GETRESULTS:
+	case GetResults:
 		if err := returnAllResults(message); err != nil {
 			Configs.Logger.Printf("there was an error: %#v", err)
 			_ = message.Conn.WriteJSON(models.NewErrorResponse(err.Error()))
 		}
 	}
 
 }
+func handleAdminBanUser(room *models.Room, message *models.UserResponse) {
```

Comment:
Pretty minor, but I wonder if this could be made less indent-y by using guard clauses.

Reply:
I completely agree with you, and I’ll put that on my backlog along with some other general cleanup work.